### ISSUE TITLE
Distinguish Folder Icon From Application Icon

### ISSUE THREAD

**Issue by pkrolkgp** [nextcloud/desktop#2903]
I would like enchancement that distinguish betwen app and nextcloud folder, because right now they look the same.

### Expected behaviour
User should know what is gonna open

### Actual behaviour
You cant tell if u open app or folder
![obraz](https://user-images.githubusercontent.com/32438012/106886965-cba38500-66e4-11eb-84c5-25d3408dc569.png)
![obraz](https://user-images.githubusercontent.com/32438012/106887169-11f8e400-66e5-11eb-835f-44e18c5c6855.png)

### Steps to reproduce
1. open start menu on windows and add to metro menu nextcloud folder and nextcloud app

### Client configuration
Client version: 3.1.2 

Operating system:
Windows

OS language:
Irrelevant

Installation path of client:
Standard

**Comment by er-vin** [comment]
@jancborchardt what do you think?

**Comment by elsiehupp** [comment]
I was actually giving some thought to [folder icons](https://github.com/nextcloud/desktop/issues/2950) just now, and it does seem like it would be reasonable for them to both be themed and be distinct from the application icon. I'm not familiar with Windows 10 icon-design norms, though.

**Comment by elsiehupp** [comment]
FWIW it could also be nice for Nextcloud to use a full-bleed Start Tile icon, if those are still a thing.

**Comment by elsiehupp** [comment]
I decided to do some quick folder-icon mockups for a Nextcloud folder on Windows 10. I was able to extract two different folder icons from a Windows DLL, and all I did was recolor them and on one of them add a Nextcloud logo. Because of copyright and whatnot, it might be preferable to do a custom SVG icon set based on more or less the same outline, with the added benefit of easier theming. (The only common image format that supports "recolor" layers is PSD, which isn't FOSS and isn't particularly easy to manipulate programmatically.) Anyway, here are the extremely preliminary mockups:
![flat-256x256](https://user-images.githubusercontent.com/9206310/109428878-ff618a00-79c6-11eb-8207-d9d7bef4a12b.png)![open-256x256](https://user-images.githubusercontent.com/9206310/109428882-02f51100-79c7-11eb-9873-e768475cb352.png)
One disadvantage of using a recolor layer (versus drawing the entire thing in vector) is that it's harder to match the exact color tones of the existing application icon. i.e. these icons are a bit high on luminosity compared to the Nextcloud app icon, because they derive their luminosity from the folder icon. If there's interest I could go ahead and do an SVG of one or both of these.

Again, another advantage with SVG is that it would be somewhat easier to programmatically theme the icon based on an OEM's brand colors and/or symbolic logo. Linux Mint has [a Python script for rasterizing SVG icons](https://github.com/linuxmint/mint-y-icons/pull/285), and I imagine it wouldn't be too difficult to adapt to spit out Windows ICO files instead. I imagine the color fill and possibly even the logo could be added using an additional script, though that's a bit beyond the scope at the moment.

One other thing worth mentioning is that it's common for smaller sizes of icons to use simpler glyphs. So the smaller sizes of the Nextcloud folder icon might just be the Nextcloud logo itself, rather than a folder or even a circle. See, for comparison, the OneDrive sidebar icon that is enabled by default in Windows Explorer (in contrast to the "Music" icon above it, which for whatever reason still features the folder symbol):

![onedrive-sidebar](https://user-images.githubusercontent.com/9206310/109429237-ac88d200-79c8-11eb-9feb-820e3085d198.jpg)

**Comment by elsiehupp** [comment]
As for Start Menu Tile icons, it looks like they're part of the [VisualElementsManifest](https://github.com/nextcloud/desktop/blob/4ea64bf9ee5a046b8210eed70fd38750405990b2/theme/nextcloud.VisualElementsManifest.xml):

https://docs.microsoft.com/en-us/windows/uwp/design/style/app-icons-and-logos

**Comment by elsiehupp** [comment]
I did some further digging, and [this StackOverflow answer](https://stackoverflow.com/a/60918342) explains how to set a folder icon programmatically in Windows.

The only related class in Qt appears to be [QFileIconProvider](https://doc.qt.io/qt-5/qfileiconprovider.html), but it only seems to apply to icons within the application, not icons as they appear in the operating system's file browser.

**Comment by elsiehupp** [comment]
It seems like the easiest way to use a folder icon would be to add it as the second icon in the `.exe` and then change line 50 to the following:
```
desktopIni.write(",1\r\n");
```
The second icon would have to be added somewhere in [AddAppIconMacro.cmake](https://github.com/nextcloud/desktop/blob/b79a82cc3f99f867e1a6e19f3babef0fa2fde01f/cmake/modules/AddAppIconMacro.cmake). I can dig further.

**Comment by jancborchardt** [comment]
@elsiehupp how do others like Dropbox and Google Drive solve this issue on Windows, which kind of separate icons do they have?

**Comment by er-vin** [comment]
> It seems like the easiest way to use a folder icon would be to add it as the second icon in the `.exe` and then change line 50 to the following:
> 
> ```
> desktopIni.write(",1\r\n");
> ```

Yes, looks like you're on the right track there.

> The second icon would have to be added somewhere in [AddAppIconMacro.cmake](https://github.com/nextcloud/desktop/blob/b79a82cc3f99f867e1a6e19f3babef0fa2fde01f/cmake/modules/AddAppIconMacro.cmake). I can dig further.

Yep, I think this macro is a bit moody though. Be careful, here lie dragons. ;-)

**Comment by elsiehupp** [comment]
I had been working on a Nextcloud folder-icon set based on the Windows 10 folder icons, but then [Microsoft up and released a major update to the Windows 10 icon theme](https://www.theverge.com/2021/3/24/22348639/microsoft-windows-10-file-explorer-new-icons-system-update):

![Old and New Windows 10 Pictures Folder Icons](https://user-images.githubusercontent.com/9206310/116815163-4fc1a900-ab2a-11eb-9f29-0fa69d588134.jpg)

I have not been able to get Windows installed on my computer—the Windows installer doesn’t like my partitioning scheme—so I don’t have an easy way of accessing [the DLLs containing the new icons](https://www.digitalcitizen.life/where-find-most-windows-10s-native-icons/) for the purpose of extracting them.

If any of you have a working copy of Windows with the new icons installed, could you try and get me a copy of the new `%systemroot%\system32\shell32.dll`? Or maybe figure out if the icons are located somewhere else? Thanks!

**Comment by pkrolkgp** [comment]
I suggest You to get official microsoft VM's.
[Download](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/)

**Comment by elsiehupp** [comment]
For a variety of reasons I would prefer to get an actual normal Windows 10 install up and running. I did get a Windows 10 VM running previously, but among other things it didn’t automatically activate with my hardware profile. (My main Linux PC shipped with Windows preinstalled, so I have an OEM license with no printed key.)

**Comment by elsiehupp** [comment]
> Hi @jancborchardt—Sorry for the lack of explanation! I closed this PR partially because of the lack of activity, yes, but also because Microsoft no longer uses Start Menu Tiles in Windows 11.
> 
> There’s a new feature in Windows 11 called [“widgets”](https://support.microsoft.com/en-us/windows/stay-up-to-date-with-widgets-7ba79aaa-dac6-4687-b460-ad16a06be6e4) which seem similar to [Live Tiles](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/creating-tiles), but it’s not clear how to create them. It could be nice to have a Nextcloud widget and a Nextcloud Live Tile that are roughly equivalent to each other, but that would probably be a lot of work, and it would have to wait on Microsoft releasing a widget development guide.
> 
> If you want to use a full-bleed Start Tile in Windows 10 regardless, it would be worth testing to make sure still Windows 11 correctly uses a standard icon in the new Start Menu. I don’t currently have a working install of Windows 10 or Windows 11, so I can’t test this myself.
> 
> _Originally posted by @elsiehupp in https://github.com/nextcloud/desktop/issues/2982#issuecomment-981744008_

Since the Pull Request with the above comment got merged, I figured here might be a better place to mention that [Microsoft recently announced the upcoming availability of a third-party API for Windows 11 Widgets](https://arstechnica.com/gadgets/2022/05/third-party-widgets-are-coming-to-windows-11-which-might-actually-make-them-useful/). This API will be based on Microsoft's cross-platform [Adaptive Cards](https://github.com/Microsoft/AdaptiveCards) platform, so it might actually be worth engaging with at some point.

Yes, Windows 11 widgets are a bit off-topic for this particular thread, but I didn't feel like creating an entirely new Issue just to follow up. (It also sounds like the "Adaptive Cards" platform might be more relevant to Nextcloud Server than Nextcloud Desktop, though any Nextcloud Adaptive Card would presumably also have platform-specific local/offline extensions.)

**Comment by elsiehupp** [comment]
Anyway, if anyone here does have an installed copy of Windows 11 that they can use to get copies of [the DLLs containing the new icon set](https://www.digitalcitizen.life/where-find-most-windows-10s-native-icons/), feel free to send me the DLLs so I can extract the folder icons to use them as design templates. (I could also probably download the DLLs from some sketchy website like I ended up doing with the Windows 10 ones, but I haven't really looked into it since Windows 11 was widely released.)

**Comment by elsiehupp** [comment]
I just found [a set of extracted icons from Windows 11](https://www.reddit.com/r/Windows11/comments/o42srx/some_windows_11_icon_files/), so that should be a good starting place for this!

**Comment by Rello** [comment]
Hello,

thank you for opening the request.
The client has improved a lot since the ticket was opened.
Can you support us and check if it is still relevant? In this case we appreciate a new ticket or a comment here.
Thank you for your support
